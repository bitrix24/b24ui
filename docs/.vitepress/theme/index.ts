/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
