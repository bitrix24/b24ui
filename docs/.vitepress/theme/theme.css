/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
