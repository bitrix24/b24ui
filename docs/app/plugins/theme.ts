export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
