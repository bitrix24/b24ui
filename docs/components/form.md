---
title: Form
description: A form component designed for validation and handling submissions.
outline: deep
---
<script setup>
import FormExample from '/examples/form/FormExample.vue';
import ElementsExample from '/examples/form/FormExampleElementsWrap.vue';
import OnErrorExample from '/examples/form/FormExampleOnError.vue';
import NestedExample from '/examples/form/FormExampleNested.vue';
import NestedListExample from '/examples/form/FormExampleNestedList.vue';
</script>
# Form

::: warning We are still updating this page
Some data may be missing here â€” we will complete it shortly.
:::

<Description
  nuxt-ui="https://ui3.nuxt.dev/components/form"
  git="https://github.com/bitrix24/b24ui/blob/main/src/runtime/components/Form.vue"
  demo="/components/form"
>
  A form component designed for validation and handling submissions.
</Description>

## Usage

Use the Form component to validate form data using validation libraries such as [Valibot](https://github.com/fabian-hiller/valibot), [Zod](https://github.com/colinhacks/zod), [Yup](https://github.com/jquense/yup), [Joi](https://github.com/hapijs/joi), [Superstruct](https://github.com/ianstormtaylor/superstruct) or your own validation logic.

It works with the [FormField](/components/form-field) component to display error messages around form elements automatically.

### Schema Validation

It requires two props:

- `state` - a reactive object holding the form's state.
- `schema` - any [Standard Schema](https://standardschema.dev/) or a schema from [Yup](https://github.com/jquense/yup), [Joi](https://github.com/hapijs/joi) or [Superstruct](https://github.com/ianstormtaylor/superstruct).

::: warning
**No validation library is included** by default, ensure you **install the one you need**.
:::

Errors are reported directly to the [FormField](/components/form-field) component based on the `name` or `error-pattern` prop. This means the validation rules defined for the `email` attribute in your schema will be applied to `<FormField name="email">`{lang="vue"}.

Nested validation rules are handled using dot notation. For example, a rule like `{ user: z.object({ email: z.string() }) }`{lang="ts"} will be applied to `<FormField name="user.email">`{lang="vue"}.

### Custom Validation

Use the `validate` prop to apply your own validation logic.

The validation function must return a list of errors with the following attributes:

- `message` - the error message to display.
- `name` - the `name` of the `FormField` to send the error to.

::: info
It can be used alongside the `schema` prop to handle complex use cases.
:::

<div>
  <ClientOnly>
    <ComponentShowExample>
      <FormExample base="/b24ui" class="w-full sm:w-2/3" />
    </ComponentShowExample>
  </ClientOnly>
</div>

::: details
<<< @/examples/form/FormExample.vue{vue:line-numbers}
:::

### Input Events

The Form component automatically triggers validation when an input emits an `input`, `change`, or `blur` event.

- Validation on `input` occurs **as you type**.
- Validation on `change` occurs when you **commit to a value**.
- Validation on `blur` happens when an input **loses focus**.

You can control when validation happens this using the `validate-on` prop.

::: tip
The form always validates on submit.
:::

::: tip
You can use the [`useFormField`](/components/composables/use-form-field) composable to implement this inside your own components.
:::

<div>
  <ClientOnly>
      <ElementsExample />
  </ClientOnly>
</div>

::: details
<<< @/examples/form/FormExampleElements.vue{vue:line-numbers}
:::

### Error Event

You can listen to the `@error` event to handle errors. This event is triggered when the form is submitted and contains an array of `FormError` objects with the following fields:

- `id` - the input's `id`.
- `name` - the `name` of the `FormField`
- `message` - the error message to display.

Here's an example that focuses the first input element with an error after the form is submitted:

<div>
  <ClientOnly>
    <ComponentShowExample>
      <OnErrorExample class="w-full sm:w-2/3" />
    </ComponentShowExample>
  </ClientOnly>
</div>

::: details
<<< @/examples/form/FormExampleOnError.vue{vue:line-numbers}
:::

### Nesting Forms

Nesting form components allows you to manage complex data structures, such as lists or conditional fields, more efficiently.

For example, it can be used to dynamically add fields based on user's input:

<div>
  <ClientOnly>
    <ComponentShowExample>
      <NestedExample class="w-full sm:w-2/3" />
    </ComponentShowExample>
  </ClientOnly>
</div>

::: details
<<< @/examples/form/FormExampleNested.vue{vue:line-numbers}
:::

Or to validate list inputs:

<div>
  <ClientOnly>
    <ComponentShowExample>
      <NestedListExample class="w-full sm:w-2/3" />
    </ComponentShowExample>
  </ClientOnly>
</div>

::: details
<<< @/examples/form/FormExampleNestedList.vue{vue:line-numbers}
:::

## API

### Props

<ComponentProps component="Form" />

### Slots

<ComponentSlots component="Form" />

### Emits

```ts
/**
 * Emitted events for the Form component
 */
interface FormEmits {
  error: (payload: [payload: FormErrorEvent]) => void;
  submit: (payload: [payload: FormSubmitEvent<any>]) => void;
}
```

### Expose

You can access the typed component instance using [`useTemplateRef`](https://vuejs.org/api/composition-api-helpers.html#usetemplateref).

```vue
<script setup lang="ts">
import { useTemplateRef } from 'vue'

const form = useTemplateRef('form')

async function makeValidate() {
  if (!form.value) {
    return
  }

  try {
    const state = await form.value.validate({
      silent: false
    })

    console.log(state)
  } catch (error) {
    console.log(
      `Some error ${error}`,
      error?.formId
        ? {
          formId: error.formId,
          errors: error?.errors
        }
        : '?'
    )
  }
}

function makeClear() {
  if (!form.value) {
    return
  }

  form.value.clear()
}
</script>

<template>
  <B24Form ref="form" />

  <div class="flex flex-row gap-4 items-center justify-between">
    <B24Button
      label="Validate"
      @click="makeValidate"
    />
    <B24Button
      label="Clear"
      @click="makeClear"
    />
  </div>
</template>
```

This will give you access to the following:

| Name                                                                                                                    | Type                                                                                                                                                     |
|-------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
| `submit()`{lang="ts"}                                                                                                   | `Promise<void>`{lang="ts"} <br> <p>Triggers form submission.</p>                                                                                         |
| `validate(opts: { name?: keyof T \| (keyof T)[], silent?: boolean, nested?: boolean, transform?: boolean })`{lang="ts"} | `Promise<T>`{lang="ts"} <br> <p>Triggers form validation. Will raise any errors unless `opts.silent` is set to true.</p>                                 |
| `clear(path?: keyof T \| RegExp)`{lang="ts"}                                                                            | `void` <br> <p>Clears form errors associated with a specific path. If no path is provided, clears all form errors.</p>                                   |
| `getErrors(path?: keyof T \| RegExp)`{lang="ts"}                                                                        | `FormError[]`{lang="ts"} <br> <p>Retrieves form errors associated with a specific path. If no path is provided, returns all form errors.</p>             |
| `setErrors(errors: FormError[], name?: keyof T \| RegExp)`{lang="ts"}                                                   | `void` <br> <p>Sets form errors for a given path. If no path is provided, overrides all errors.</p>                                                      |
| `errors`{lang="ts"}                                                                                                     | `Ref<FormError[]>`{lang="ts"} <br> <p>A reference to the array containing validation errors. Use this to access or manipulate the error information.</p> |
| `disabled`{lang="ts"}                                                                                                   | `Ref<boolean>`{lang="ts"}                                                                                                                                |
| `dirty`{lang="ts"}                                                                                                      | `Ref<boolean>`{lang="ts"} `true` if at least one form field has been updated by the user.                                                                |
| `dirtyFields`{lang="ts"}                                                                                                | `DeepReadonly<Set<keyof T>>`{lang="ts"} Tracks fields that have been modified by the user.                                                               |
| `touchedFields`{lang="ts"}                                                                                              | `DeepReadonly<Set<keyof T>>`{lang="ts"} Tracks fields that the user interacted with.                                                                     |
| `blurredFields`{lang="ts"}                                                                                              | `DeepReadonly<Set<keyof T>>`{lang="ts"} Tracks fields blurred by the user.                                                                               |
