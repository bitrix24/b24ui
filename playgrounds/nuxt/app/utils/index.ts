import { splitByCase, upperFirst } from 'scule'

export function upperName(name: string) {
  return splitByCase(name).map(p => upperFirst(p)).join('')
}
