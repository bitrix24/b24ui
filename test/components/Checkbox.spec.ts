import { describe, it, expect, test } from 'vitest'
import Checkbox from '../../src/runtime/components/Checkbox.vue'
import type { CheckboxProps, CheckboxSlots } from '../../src/runtime/components/Checkbox.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/checkbox'
import { renderForm } from '../utils/form'
import { mount, flushPromises } from '@vue/test-utils'
import type { FormInputEvents } from '../../src/module'

describe('Checkbox', () => {
  const sizes = Object.keys(theme.variants.size) as any

  it.each([
    // Props
    ['with modelValue', { props: { modelValue: true } }],
    ['with defaultValue', { props: { defaultValue: true } }],
    ['with id', { props: { id: 'id' } }],
    ['with name', { props: { name: 'name' } }],
    ['with value', { props: { value: 'value' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with indeterminate', { props: { defaultValue: 'indeterminate' } }],
    ['with label', { props: { label: 'Label' } }],
    ['with required', { props: { label: 'Label', required: true } }],
    ['with description', { props: { label: 'Label', description: 'Description' } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { size } }]),
    ['with color air-primary-success', { props: { color: 'air-primary-success', defaultValue: true } }],
    ['with ariaLabel', { attrs: { 'aria-label': 'Aria label' } }],
    ['with as', { props: { as: 'section' } }],
    ['with class', { props: { class: 'inline-flex' } }],
    ['with b24ui', { props: { b24ui: { wrapper: 'ms-4' } } }],
    // Slots
    ['with label slot', { slots: { label: () => 'Label slot' } }],
    ['with description slot', { slots: { label: () => 'Description slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: CheckboxProps, slots?: Partial<CheckboxSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, Checkbox)
    expect(html).toMatchSnapshot()
  })

  describe('emits', () => {
    test('update:modelValue event', async () => {
      const wrapper = mount(Checkbox)
      const input = wrapper.findComponent({ name: 'CheckboxRoot' })
      await input.vm.$emit('update:modelValue', true)
      expect(wrapper.emitted()).toMatchObject({ 'update:modelValue': [[true]] })
    })

    test('change event', async () => {
      const wrapper = mount(Checkbox)
      const input = wrapper.findComponent({ name: 'CheckboxRoot' })
      await input.vm.$emit('update:modelValue', false)
      expect(wrapper.emitted()).toMatchObject({ change: [[{ type: 'change' }]] })
    })
  })

  describe('form integration', async () => {
    async function createForm(validateOn?: FormInputEvents[]) {
      const wrapper = await renderForm({
        props: {
          validateOn,
          validateOnInputDelay: 0,
          async validate(state: any) {
            if (!state.value)
              return [{ name: 'value', message: 'Error message' }]
            return []
          }
        },
        slotTemplate: `
        <B24FormField name="value">
          <B24Checkbox v-model="state.value" />
        </B24FormField>
        `
      })
      const input = wrapper.findComponent({ name: 'CheckboxRoot' })
      return {
        wrapper,
        input
      }
    }

    test('validate on change works', async () => {
      const { input, wrapper } = await createForm(['change'])
      await input.vm.$emit('update:modelValue', false)
      await flushPromises()

      expect(wrapper.text()).toContain('Error message')

      await input.vm.$emit('update:modelValue', true)
      await flushPromises()
      expect(wrapper.text()).not.toContain('Error message')
    })

    test('validate on input works', async () => {
      const { input, wrapper } = await createForm(['input'])
      await input.vm.$emit('update:modelValue', false)
      await flushPromises()

      expect(wrapper.text()).toContain('Error message')

      await input.vm.$emit('update:modelValue', true)
      await flushPromises()
      expect(wrapper.text()).not.toContain('Error message')
    })
  })
})
