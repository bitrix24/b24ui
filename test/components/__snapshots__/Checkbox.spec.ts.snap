// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > renders with as correctly 1`] = `
"<section class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</section>"
`;

exports[`Checkbox > renders with b24ui correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with class correctly 1`] = `
"<div class="relative items-start inline-flex">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with color success correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white ring-inset focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-green-500 dark:focus-visible:outline-green-600 size-4 ring-2 ring-green-500 bg-green-500 dark:ring-green-600 dark:bg-green-600 dark:text-green-250" id="v-0-0" role="checkbox" type="button" aria-checked="true" aria-required="false" data-state="checked"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-full" style="pointer-events: none;">
        <path fill="currentColor" fill-rule="evenodd" d="M9.602 19.018a.3.3 0 0 1-.419 0l-5.851-5.703a.4.4 0 0 1 0-.573l1.65-1.609a.4.4 0 0 1 .56 0l3.85 3.754L18.46 6.05a.4.4 0 0 1 .558 0l1.651 1.61a.4.4 0 0 1 0 .572z" clip-rule="evenodd"></path>
      </svg>
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with defaultValue correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white ring-inset focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4 ring-2 ring-blue-500 bg-blue-500 dark:ring-blue-600 dark:bg-blue-600 dark:text-blue-250" id="v-0-0" role="checkbox" type="button" aria-checked="true" aria-required="false" data-state="checked"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-full" style="pointer-events: none;">
        <path fill="currentColor" fill-rule="evenodd" d="M9.602 19.018a.3.3 0 0 1-.419 0l-5.851-5.703a.4.4 0 0 1 0-.573l1.65-1.609a.4.4 0 0 1 .56 0l3.85 3.754L18.46 6.05a.4.4 0 0 1 .558 0l1.651 1.61a.4.4 0 0 1 0 .572z" clip-rule="evenodd"></path>
      </svg>
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with description correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <div class="font-b24-primary ms-2 text-md"><label for="v-0-0" class="cursor-pointer block font-regular text-base-master dark:text-base-400 leading-5">Label</label>
    <p class="text-base-500 dark:text-base-600">Description</p>
  </div>
</div>"
`;

exports[`Checkbox > renders with description slot correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <div class="font-b24-primary ms-2 text-md"><label for="v-0-0" class="cursor-pointer block font-regular text-base-master dark:text-base-400 leading-5">Description slot</label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Checkbox > renders with disabled correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4 cursor-not-allowed opacity-75" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked" data-disabled="" disabled="">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with id correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="id" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with indeterminate correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white ring-inset focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4 ring-2 ring-blue-500 bg-blue-500 dark:ring-blue-600 dark:bg-blue-600 dark:text-blue-250" id="v-0-0" role="checkbox" type="button" aria-checked="mixed" aria-required="false" data-state="indeterminate"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-full" style="pointer-events: none;">
        <path fill="currentColor" fill-rule="evenodd" d="M7 11v2h10v-2H7" clip-rule="evenodd"></path>
      </svg>
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with label correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <div class="font-b24-primary ms-2 text-md"><label for="v-0-0" class="cursor-pointer block font-regular text-base-master dark:text-base-400 leading-5">Label</label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Checkbox > renders with label slot correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <div class="font-b24-primary ms-2 text-md"><label for="v-0-0" class="cursor-pointer block font-regular text-base-master dark:text-base-400 leading-5">Label slot</label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Checkbox > renders with modelValue correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white ring-inset focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4 ring-2 ring-blue-500 bg-blue-500 dark:ring-blue-600 dark:bg-blue-600 dark:text-blue-250" id="v-0-0" role="checkbox" type="button" aria-checked="true" aria-required="false" data-state="checked"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-full" style="pointer-events: none;">
        <path fill="currentColor" fill-rule="evenodd" d="M9.602 19.018a.3.3 0 0 1-.419 0l-5.851-5.703a.4.4 0 0 1 0-.573l1.65-1.609a.4.4 0 0 1 .56 0l3.85 3.754L18.46 6.05a.4.4 0 0 1 .558 0l1.651 1.61a.4.4 0 0 1 0 .572z" clip-rule="evenodd"></path>
      </svg>
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with name correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with required correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="true" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <div class="font-b24-primary ms-2 text-md"><label for="v-0-0" class="cursor-pointer block font-regular text-base-master dark:text-base-400 leading-5 after:content-['*'] after:ms-0.5 after:text-red-500 dark:after:text-red-600">Label</label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Checkbox > renders with size lg correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-5" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with size md correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with size sm correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-4"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-3.5" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with size xs correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-4"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-3" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with value correctly 1`] = `
"<div class="relative flex items-start">
  <div class="flex items-center h-5"><button class="cursor-pointer shrink-0 flex items-center justify-center rounded-2xs text-white dark:text-base-150 ring ring-inset ring-base-300 dark:ring-base-700 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 dark:focus-visible:outline-blue-600 size-4" id="v-0-0" role="checkbox" type="button" aria-checked="false" aria-required="false" data-state="unchecked">
      <!---->
      <!---->
    </button></div>
  <!--v-if-->
</div>"
`;
